### DISCUSSION TITLE
[V3] error: Only arrays and Traversables can be unpacked

### DISCUSSION THREAD

**Discussion by polycode-nz** [livewire/livewire#6856] [1. Help] [answered]
I'm not sure where to start with this... Battling upgrading a project to 3.

I'm trying to listen to an event in the component fired from js and I get a PHP type error: "Only arrays and Traversables can be unpacked".

It doesn't appear to get as far as my component code.

However the XHR request is also not what (where) I'd expect. It goes it `livewire/update`.

The blade/js code:

```
@push('scripts')
   <script type="text/javascript">
      document.addEventListener('livewire:initialized', () => {
         window.emitEditClick = () => {
            @this.dispatchSelf('edit-that', clickedID);
         };
      });
   </script>
@endpush
```

That results in an XHR call to: http://local.somesite/livewire/update with the error.

Potentially interesting, if I change `edit-that` above to something non-existent `edit-that-potato` nothing at all happens.

In the component I have a public function:

```
#[On('edit-that')] 
public function editSomething(int $id)
{
   echo 'editSomething';
   die;
}
```

From the error screen, the last vendor frame is `Livewire\Features\SupportEvents\ SupportEvents` line 29:

```
$returnEarly(
   wrap($this->component)->$method(...$params)
);
```
 
Any ideas what I'm missing? Happy to share more info such as:

- I have `'legacy_model_binding' => true,` set
- This is a full page component with a route like: `Route::get('/configuration/accidents/age-ranges', AccidentAgeRangeManager::class)->name('config.accidents.age-ranges');`
- Laravel version: 10.23.1
- Livewire version: 3.0.5
- PHP version: 8.1.14

**Comment by QuentinGab** [comment, marked as answer]
This error usually occur when livewire can't figure out how to convert your variables from JS to PHP. 
In Livewire 3, parameters of component methods must be named, so when dispatching events, you have to use an object in which each key refers to the php variable name.

You might want to try

```js
@this.dispatchSelf('edit-that', { id: clickedID} );
```

**Reply by polycode-nz** [reply]
Thanks for the reply. I could've sworn I tried that!

**Reply by SamirMaharjan** [reply]
this worked for me. really helped me alot. thanks

**Reply by chepkosum** [reply]
Can someone help me with this? im getting the same error of Only arrays and Traversables can be unpacked

here is a snippet of the code .

            onApprove: (data, actions) => {
                return actions.order.capture().then(function(orderData) {
                    //Successful capture! For dev/demo purposes:
                    console.log('Capture result', orderData, JSON.stringify(orderData, null, 2));
                    const transaction = orderData.purchase_units[0].payments.captures[0];
                    if (transaction.status == "COMPLETED") {
                        Livewire.dispatch('transactionEmit', {id: transaction.id});
                    }
                    // alert(`Transaction) ${transaction.status}: ${transaction.id}\n\nSee console for all available details`);
                    //When ready to go live, remove the alert and show a success message within this page. For exampl:
                    //const element = document.getElementById('paypal-button-container');
                    // element.innerHTML = '<h3>Thank you for your payments!</h3>';
                    // or go to another URL: actions.redirect('thank_you.html');

                });
            }

**Reply by bennettblack** [reply]
Bless you.

**Reply by Icrmsoftware** [reply]
Thanks for this!

**Reply by v-e-y** [reply]
In my case (Select2) when I was migrate code from LW v2 I just wrap values in to array `[values]`

```javascript
$wire.dispatch('{{ $this->selectedEventName }}', [values]);
```

**Reply by lucas-dos-santos-silva** [reply]
@v-e-y in my case only add [] on value works fine
